docs/script_matching.md

```diff
@@ -0,0 +1,93 @@
+One central task for skribo is choosing a font most suitable for the requested script. The requested script is a combination of an explicit lang tag and locale system settings.
+
+Part of what makes this requirement tricky is that there is a fuzzy line between the scope of skribo and the scope of [font-kit]. Conceptually, skribo should be a consumer of font metadata, and font-kit is a provider. At a finer grain, there are decisions about exactly what that interface should look like. The main purpose of this document is to describe the problems to be solved.
+
+## Background
+
+Selection of font is not strictly based on Unicode coverage, but is also affected by locale. The biggest impact is on Han unification, but there are other examples. Suitability of a font for a script is *not* encoded in the font's tables, but is metadata about the system's font configuration. Mechanisms for accessing this metadata vary widely from platform to platform.
+
+A significant amount of research for this topic was reading through the [Skia] code, as that's the platform abstraction used by Chromium, so can be used as a rough first draft of requirements for correct cross-platform text layout for the Web. I also used Gecko and Qt as sources.
+
+The main focus of this document is font fallback, but a related platform-specific concern is resolution of generic aliases (system UI font, generic serif, etc). This may be out of scope for skribo itself, but likely within the scope of font-kit. More clarity on requirements for each crate is needed.
+
+TODO: write a clear requirement, link to this section of requirements doc.
+
+## BCP-47
+
+I strongly recommend the use of BCP-47 as the identifier for language, script, and other locale metadata. This is an easy decision for web use cases, as it is the standard for the [lang] tag. The main challenge is that mechanisms for system font metadata in general predate BCP-47, so there will be some impedance matching.
+
+TODO: investigate Rust ecosystem for common BCP-47 tag representation.
+
+## Platform-specific considerations
+
+Each platform handles access to metadata (including script) in a different way. Platforms generally treat their text stack as an opaque black box, and vary in how much low-level access they give.
+
+### Fontconfig
+
+The [Fontconfig] configuration file format specifies "langset" as an an "RFC-3066-style" language. [RFC 3066] is a predecessor to BCP-47 (dated 2001), and basically specifies language and country, with no provision for explicit script or variant. For the purpose of Han unification, the convention is to infer script from country. For example, "zh-CN" could be translated to "zh-Hans", "zh-TW" to "zh-Hant". However, after a little investigation, it's not clear to me how useful it is to do sophisticated processing here, as the default fontconfig on a clean Debian 9 install lists doesn't specify "langset" attributes, but just has a few informal descriptions in comments.
```

Comment:
IIRC a lot of systems _don't_ handle this so everyone just uses CN and TW. Not sure about font systems specifically.

I have had hans vs hant trigger differences in browsers

Reply:
I think this is just the march of progress, and hopefully in a few decades the use of country to represent script will fade away.